drivers/android/context.rs

```diff
@@ -24,24 +27,23 @@ unsafe impl Send for Context {}
 unsafe impl Sync for Context {}
 
 impl Context {
-    pub(crate) fn new() -> Result<Pin<Arc<Self>>> {
-        let mut ctx_ref = Arc::try_new(Self {
-            // SAFETY: Init is called below.
-            manager: unsafe {
-                Mutex::new(Manager {
-                    node: None,
-                    uid: None,
-                })
+    pub(crate) fn new() -> Result<Ref<Self>> {
```

Comment:
I notice you're replacing `Pin<Arc<T>>` with `Ref<T>` here. This is of course correct, since `Ref<T>` is now "intrinsically pinned" (or whatever name we want to give it).

Would you perhaps be open to one of the following (or both):
1. replace `Pin<Ref<T>>` with `Ref<T>` everywhere
2. rename `Ref<T>` to `PinRef<T>` or `PinnedRef<T>` or some such

(this should of course not hold up this PR)

Reply:
> Would you perhaps be open to one of the following (or both):
> 
> 1. replace `Pin<Ref<T>>` with `Ref<T>` everywhere

I will consider this as I switch users of `Arc` to `Ref`. I don't anticipate any challenges but I do run into them I'll rope you in.

> 2. rename `Ref<T>` to `PinRef<T>` or `PinnedRef<T>` or some such

I'm open to this, but I'm not convinced yet that it's a good idea. Mostly because being pinned is a property of `Ref`, but it's not a defining one (being ref-counted is).